### PR TITLE
Add ProjectResourceStore interface

### PR DESCRIPTION
**Pull request by kohidave**
This interface describes the methods to fetch resources for a particular
project.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/archer/project.go [hunk 1]
```diff
@@ -31,3 +39,11 @@ type ProjectCreator interface {
 type ProjectGetter interface {
 	GetProject(projectName string) (*Project, error)
 }
+
+// ProjectResourceStore fetches resources related to the project.
+type ProjectResourceStore interface {
+	// Fetch all regional resources in a project.
+	GetRegionalProjectResources(project *Project) ([]*ProjectRegionalResources, error)
+	// Fetch the project regional resources for a particular region
```
Comment (efekarakus) [inline internal/pkg/archer/project.go]: nit: can we have all comments end with a dot

Comment (SoManyHs) [reply internal/pkg/archer/project.go]: is that a style thing?

File: internal/pkg/archer/project.go [hunk 2]
```diff
@@ -7,7 +7,15 @@ package archer
...
-	Version   string `json:"version"` // The version of the project layout in the underyling datastore (e.g. SSM)
+	Version   string `json:"version"` // The version of the project layout in the underlying datastore (e.g. SSM)
+}
+
+// ProjectRegionalResources represent project resources that are regional
+type ProjectRegionalResources struct {
+	Region         string            // The region these resources are in
+	KMSKeyARN      string            // A KMS Key ARN for encrypting Pipeline artifacts
+	S3Bucket       string            // S3 bucket for Pipeline artifacts
+	RepositoryURLs map[string]string // The image repository URLs by app name
```
Comment (SoManyHs) [inline internal/pkg/archer/project.go]: Question for my own edification: are we only ever going to support ECR repositories?

File: internal/pkg/archer/project.go (hunk 1 above)
Comment (efekarakus) [reply internal/pkg/archer/project.go]: Yeah: https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences

> Comments should begin with the name of the thing being described and end in a period

File: internal/pkg/archer/project.go (hunk 2 above)
Comment (kohidave) [reply internal/pkg/archer/project.go]: That's right. But I figured I'd keep that detail out of there as much as I can.

File: internal/pkg/archer/project.go (hunk 1 above)
Comment (kohidave) [reply internal/pkg/archer/project.go]: WHAT IF THE COMMENT IS A QUANDARY.

**Pull request merged**